nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -412,6 +412,21 @@ sub filterUnits {
 # Make systemd reload its units.
 system("@systemd@/bin/systemctl", "daemon-reload") == 0 or $res = 3;
 
+# Reload user units
+opendir (DIR, "/run/user") or $res = 3;
+while (my $f = readdir(DIR)) {
+    if ($f =~ "\\.") {
+        next;
+    }
+
+    my $name = getpwuid(int($f));
+    print STDERR "reloading user units for $name...\n";
+
+    system("su -l $name -c \$'XDG_RUNTIME_DIR=/run/user/$f @systemd@/bin/systemctl --user daemon-reload'");
```

Comment:
Running programs under arbitrary user accounts from a root script frightens me. In any case, it should be 
```
system("su", "-l", $name, "-c" "command...");
```
to prevent shell interpolation etc.

Reply:
thx, totally missed that!